Configurateur/front/lib/components/custom_footer.dart

```diff
@@ -0,0 +1,149 @@
+// ignore_for_file: unrelated_type_equality_checks, use_build_context_synchronously
+
+import 'package:flutter/material.dart';
+import 'package:footer/footer.dart';
+import 'package:front/services/storage_service.dart';
+import 'package:front/services/theme_service.dart';
+import 'package:front/styles/themes.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+
+class CustomFooter extends StatefulWidget {
+  const CustomFooter({super.key, required BuildContext context});
+
+  @override
+  State<CustomFooter> createState() => CustomFooterState();
+}
+
+class CustomFooterState extends State<CustomFooter> {
+  String? token = '';
+  String? userMail = '';
+
+  void checkToken() async {
+    token = await storageService.readStorage('token');
+    storageService.getUserMail().then((value) => userMail = value);
+    setState(() {});
+  }
+
+  void goToCreation() async {
+    if (await storageService.readStorage('token') == '') {
+      context.go("/login");
+    } else {
+      context.go("/container-creation/");
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    checkToken();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Footer(
+      backgroundColor: const Color(0xffFEDC97),
+      child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          children: <Widget>[
+            Center(
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: <Widget>[
+                  SizedBox(
+                      height: 45.0,
+                      width: 45.0,
+                      child: Center(
+                        child: Card(
+                          elevation: 5.0,
+                          shape: RoundedRectangleBorder(
+                            borderRadius: BorderRadius.circular(
+                                25.0), // half of height and width of Image
+                          ),
+                          child: IconButton(
+                            tooltip: "Page de contact",
+                            icon: const Icon(
+                              Icons.contact_page,
+                              size: 20.0,
+                            ),
+                            color: Provider.of<ThemeService>(context).isDark
+                                ? darkTheme.primaryColor
+                                : lightTheme.primaryColor,
+                            onPressed: () {
+                              context.go("/contact");
+                            },
+                          ),
+                        ),
+                      )),
```

Comment:
Maybe add , after the first ) so that the file is more readable

Reply:
done